enterprise/server/remote_execution/runner/runner.go

```diff
@@ -614,34 +647,64 @@ func (p *Pool) take(q *query) *CommandRunner {
 			metrics.RecycleRunnerRequestStatusLabel: hitStatusLabel,
 		}).Inc()
 
-		return r
+		return r, nil
 	}
 
 	metrics.RecycleRunnerRequests.With(prometheus.Labels{
 		metrics.RecycleRunnerRequestStatusLabel: missStatusLabel,
 	}).Inc()
 
-	return nil
+	return nil, nil
 }
 
-// Size returns the current number of paused runners in the pool.
-func (p *Pool) Size() int {
+// RunnerCount returns the total number of runners in the pool.
+func (p *Pool) RunnerCount() int {
 	p.mu.RLock()
 	defer p.mu.RUnlock()
 	return len(p.runners)
 }
 
+// PausedRunnerCount returns the current number of paused runners in the pool.
+func (p *Pool) PausedRunnerCount() int {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	return p.pausedRunnerCount()
+}
+
+// ActiveRunnerCount returns the number of non-paused runners in the pool.
+func (p *Pool) ActiveRunnerCount() int {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	n := 0
+	for _, r := range p.runners {
+		if r.state != paused {
+			n++
+		}
+	}
+	return n
+}
+
+func (p *Pool) pausedRunnerCount() int {
+	n := 0
+	for _, r := range p.runners {
+		if r.state == paused {
+			n++
+		}
+	}
+	return n
+}
+
```

Comment:
maybe a shorter way to do this is to have a function that locks mu and counts the runners in a particular state:

func countRunnersInState(s state) {
 mu.Lock()
 defer mu.Unlock()
 count := 0
 for _, r := p.runners {
    if r.state == s {
      count += 1
   }
  }
  return count
}

then Active / Paused / Running are just:
func ActiveRunnerCount() int {
  return countRunnersInState(ACTIVE)
}
... etc ...

Reply:
Shortened to `len(p.runners) - p.pausedRunnerCount()` since "active" is not an actual state (it's defined as "not paused")